### ISSUE TITLE
Unable to reference Javascript files in build

### ISSUE THREAD

**Issue by soumyamondal** [mingyaulee/Blazor.BrowserExtension#65]
Hi,
Currently I am trying to do build a sample browser extension for outlook platform .
I need to inject a separate JS file as soon as the app.js is injected and loaded in the browser,but unfortunately the the separate script file(interop.js) is not getting injected into the DOM.
I am calling the method CreateAlert in interop. Js file from the Home razor component using  JS interop of Blazor WASM, but I am getting the below error :-
 
![image](https://github.com/mingyaulee/Blazor.BrowserExtension/assets/24987312/1f8dc32f-1c76-4a9f-aac6-ec60f01864f4)

Note:-
I have tried other approaches for referring the methods also but none of them worked:-
1. added the JS methods of interop.js  into app.js file directly and wrapped them in windows.load event.
2.added the JS methods of interop,js into a separate script tag of index.html.

I want to know, in this particular project template how to I reference external JS files from the native razor components, how the connection is made?
Also how can I load separate Javascript files to DOM during runtime and call the methods in them from razor components.

Below arescreenshots of my project files:-
<img width="686" alt="image" src="https://github.com/mingyaulee/Blazor.BrowserExtension/assets/24987312/62057545-f377-476d-9d41-44bd020eca19">
<img width="665" alt="image" src="https://github.com/mingyaulee/Blazor.BrowserExtension/assets/24987312/f06c493e-820d-4942-9060-422745a8bd67">
<img width="633" alt="image" src="https://github.com/mingyaulee/Blazor.BrowserExtension/assets/24987312/353bc0a2-4d86-4a6b-98b8-7d3c655c2116">
<img width="583" alt="image" src="https://github.com/mingyaulee/Blazor.BrowserExtension/assets/24987312/23975073-1a4f-4a0c-a8f2-62198ddce7f1">
![image](https://github.com/mingyaulee/Blazor.BrowserExtension/assets/24987312/eba51a7f-911b-4ddb-a4a4-12ef3becfc16)

**Comment by soumyamondal** [comment]
[https://drive.google.com/file/d/1pShrzD0HmkcEMWjnf5aLXps_FKIUCte6/view?usp=sharing](url).
I have shared my sample project , can you please guide me how to inject the interop.js file into DOM , once the application is loaded and running?

**Comment by soumyamondal** [comment]
Hi Ming, 
please let me know, if you have any update

**Comment by mingyaulee** [comment]
Hi, I am trying to find time out of my full time work to maintain this project, so it might take a bit more time to get a response. One thing I noticed you are trying to do, is to interop with a script inserted as an element in the DOM. This is impossible for any browser extension due to security reasons. The script injected will have the functions in the context of the main window, but the content scripts have a different context. Basically, you get a different `window` object from the page script vs the content script, which means the custom functions exposed in the main window are not available in the content script. These can be found in more details either from the Chrome or MDN documentation.

In short, you can try to add the interop script into manifest content script section.

**Comment by mingyaulee** [comment]
Hi @soumyamondal, I haven't been able to find time to look at your code yet. Are you able to resolve it?

**Comment by abyss616** [comment]
> Hi @soumyamondal, I haven't been able to find time to look at your code yet. Are you able to resolve it?

Hi, I have similar problem. I want to do call Blazor method from app.js. 
I am using Microsoft recommended approach - DotNet.invokeMethodAsync("MyBlazorApp", "UpdateMessage", newMessage);

But I am getting error: DotNet is not defined. 

I am guesing it has something to do with blazor.webassembly.js, but I have inlcuded it in index.html

Not sure what else to do.

Thanks for any response

**Comment by mingyaulee** [comment]
This comes from the fact that the app.js is loaded before initialization, so the Blazor app has not been started yet.

I was thinking to switch the loading of the scripts to the standard blazor initializer which supports beforeStart and afterStart initializers. However, that would mean I need to drop support for .Net 5 since this was only introduced in .Net 6.
